sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/partition_client.hpp

```diff
@@ -102,6 +103,10 @@ namespace Azure { namespace Messaging { namespace EventHubs {
         {
           messages.push_back(Models::ReceivedEventData{message.first.Value()});
         }
+        else
```

Comment:
Another check worth adding on `m_receivers[0].WaitForIncomingMessage(context);` where the length is non-zero for m_receivers?

Reply:
The runtime library will throw an exception if the array access is out of bounds. Do we *REALLY* need an explicit check beyond the ones already built into the library?